apps/web/src/pages/brand/tabs/LayoutEditor.tsx

```diff
@@ -1,16 +1,34 @@
-import { useForm, Controller } from 'react-hook-form';
+import { useForm, Controller, useFieldArray } from 'react-hook-form';
 import { EmailCustomCodeEditor } from '../../../components/templates/email-editor/EmailCustomCodeEditor';
-import { Center, Grid } from '@mantine/core';
+import { Center, Grid, Group, Modal, Title, useMantineTheme } from '@mantine/core';
 import { ArrowLeft } from '../../../design-system/icons';
-import { Button, Checkbox, colors, Input, Text, LoadingOverlay } from '../../../design-system';
+import { Button, Checkbox, colors, Input, Text, LoadingOverlay, shadows } from '../../../design-system';
 import { useEnvController } from '../../../store/use-env-controller';
 import { errorMessage, successMessage } from '../../../utils/notifications';
-import { useMutation, useQuery, useQueryClient } from '@tanstack/react-query';
-import { createLayout, getLayoutById, updateLayoutById } from '../../../api/layouts';
-import { useEffect } from 'react';
-import { ILayoutEntity } from '@novu/shared';
+import { useQueryClient } from '@tanstack/react-query';
+import { useEffect, useMemo, useState } from 'react';
+import { parse } from '@handlebars/parser';
+import { getTemplateVariables, ITemplateVariable, isReservedVariableName } from '@novu/shared';
+
 import { QueryKeys } from '../../../api/query.keys';
+import { VariableManager } from '../../../components/templates/VariableManager';
+import { VariablesManagement } from '../../../components/templates/email-editor/variables-management/VariablesManagement';
+import { useLayoutsEditor } from '../../../api/hooks/use-layouts-editor';
 
+interface ILayoutForm {
+  content: string;
+  name: string;
+  description: string;
+  isDefault: boolean;
+  variables: ITemplateVariable[];
+}
```

Comment:
You can generate the interface from `Pick<LayoutDto, 'content' | 'name' | 'description' | 'isDefault' | 'variables'>`.